samples/create_thumbnail_from_mp4_video/GenerateThumbnailsPipeline.cpp

```diff
@@ -0,0 +1,74 @@
+
+#include "GenerateThumbnailsPipeline.h"
+#include "CudaMemCopy.h"
+#include "ExternalSinkModule.h"
+#include "FileReaderModule.h"
+#include "FileWriterModule.h"
+#include "FrameMetadata.h"
+#include "H264Decoder.h"
+#include "H264Metadata.h"
+#include "JPEGEncoderNVJPEG.h"
+#include "Mp4ReaderSource.h"
+#include "Mp4VideoMetadata.h"
+#include "ValveModule.h"
+#include <boost/test/unit_test.hpp>
+
+GenerateThumbnailsPipeline::GenerateThumbnailsPipeline()
+    : pipeLine("pipeline") {}
+
+bool GenerateThumbnailsPipeline::setUpPipeLine(
+    const std::string &videoPath, const std::string &outFolderPath) {
+  // Implementation
+
+  bool parseFS = false;
+  auto h264ImageMetadata = framemetadata_sp(new H264Metadata(0, 0));
+  auto frameType = FrameMetadata::FrameType::H264_DATA;
+  auto mp4ReaderProps =
```

Comment:
put one line comment for all modules - what does it do in the pipeline / why is it required